src/main/java/org/mule/extension/connection/ConnectionHandler.java

```diff
@@ -0,0 +1,43 @@
+package org.mule.extension.connection;
+
+/**
+ * Handles connections that are created from a config of a generic type.
+ * <p/>
+ * Implementations are expected to be reusable and thread-safe
+ *
+ * @param <Config>     the generic type of the objects to be used as configs
+ * @param <Connection> the generic type of the connections to be handled
+ * @since 1.0
+ */
+//TODO: Consider movign this to the mule-api when we actually create that artifact
+public interface ConnectionHandler<Config, Connection>
+{
+
+    /**
+     * Create a new connection from the given {@code config}.
+     * <p/>
+     * The returned connection is expected to be ready to use
+     *
+     * @param config a configuration which parametrizes the connection's creation
+     * @return a ready to use {@code Connection}
+     */
+    Connection connect(Config config);
+
+    /**
+     * Disposes the given {@code connection}, freeing all its allocated resources
+     *
+     * @param connection a non {@code null} {@code Connection}.
+     */
+    void disconnect(Connection connection);
+
+    /**
+     * Validates that the given {@code connection} is still valid.
+     * <p/>
+     * This is useful to test that a connection has not timed, crashed
+     * or remotely invalidated
+     *
+     * @param config     the {@code Config} object for which the {@code connection} was created
+     * @param connection the {@code Connection} to be tested
+     */
+    void validate(Config config, Connection connection);
```

Comment:
that depends on the type of connection and handler implementations. In some cases it might not be possible, in others it might be too expensive. If you look at the connection mgmt spec there's a whole section about that and about how to configure when/how ofter should this method be called.